### PR TITLE
feat: delete creatable items

### PR DESCRIPTION
**Pull request by yurii-korolchuk**

### CODE REVIEW

**Comment by vlacher12** [comment]
@yurii-korolchuk Do we need delete button only for creatable select?

**Comment by yurii-korolchuk** [comment]
@vlacher12 Yes, I believe so

**Comment by vlacher12** [comment]
> @vlacher12 Yes, I believe so, but do you mean on QIT or overall ?

PUI can be used anywhere, we need to foresee any usage. I'd add delete button for default SelectInput, not only for creatable
cc @lazzystep @kaliahinartem @kkurylchenko

**Comment by yurii-korolchuk** [comment]
> > @vlacher12 Yes, I believe so, but do you mean on QIT or overall ?
> 
> PUI can be used anywhere, we need to foresee any usage. I'd add delete button for default SelectInput, not only for creatable cc @lazzystep @kaliahinartem @kkurylchenko

Will add, no problem, just need some time to refactor the existing solution

File: src/components/select/select-component.tsx [hunk 1]
```diff
@@ -8,13 +8,24 @@ import { Icon, Text } from 'components';
...
       {children}
+      {/* eslint-disable-next-line no-underscore-dangle */}
+      {onDeleteItem && !data.__isNew__ && (
+        <Icon
+          icon={deleteItemIcon || Icon.icons.trash}
+          className='deleteItemIcon'
+          onClick={(e) => {
+            e.stopPropagation();
+            onDeleteItem(data);
+          }}
```
Comment (vlacher12) [inline src/components/select/select-component.tsx:26]: add an extra function `handleDelete` for this

File: src/components/select/select-component.tsx [hunk 2]
```diff
@@ -8,13 +8,24 @@ import { Icon, Text } from 'components';
...
+const CustomOption = ({ deleteItemIcon, onDeleteItem, ...props }: any): JSX.Element => {
+  const { children, isSelected, data } = props;
 
   return (
     <components.Option {...props}>
-      {isSelected && <Icon icon={Icon.icons.check} className="icon" />}
+      {isSelected && <Icon icon={Icon.icons.check} className='icon' />}
       {children}
+      {/* eslint-disable-next-line no-underscore-dangle */}
+      {onDeleteItem && !data.__isNew__ && (
```
Comment (vlacher12) [inline src/components/select/select-component.tsx:19]: deleteButtonIsShown

**Review by vlacher12 — CHANGES_REQUESTED** [review]

File: src/components/select/select-component.tsx (hunk 1 above)
Comment (yurii-korolchuk) [reply src/components/select/select-component.tsx]: done

File: src/components/select/select-component.tsx (hunk 2 above)
Comment (yurii-korolchuk) [reply src/components/select/select-component.tsx]: done as well

File: src/components/select/select-component.tsx [hunk 3]
```diff
@@ -8,13 +8,24 @@ import { Icon, Text } from 'components';
...
+  /* eslint-disable-next-line no-underscore-dangle */
+  const deleteButtonIsShown = onDeleteItem && !data.__isNew__ && !isSelected;
 
   return (
     <components.Option {...props}>
-      {isSelected && <Icon icon={Icon.icons.check} className="icon" />}
+      {isSelected && <Icon icon={Icon.icons.check} className='icon' />}
       {children}
+      {deleteButtonIsShown && (
+        <Icon icon={deleteItemIcon || Icon.icons.trash} className='deleteItemIcon' onClick={handleDelete} />
```
Comment (vlacher12) [inline src/components/select/select-component.tsx:27]: can you just set `Icon.icons.trash` as default value for `deleteItemIcon`?

**Review by vlacher12 — CHANGES_REQUESTED** [review]

**Review by vlacher12 — APPROVED** [review]

**Pull request merged**